### PR TITLE
TST: add message check to pytest.raises (tests/arrays/test_boolean.py)

### PR DESCRIPTION
**Pull request by ShilpaSugan**
- [x] ref #30999 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] Added in an error message

### CODE REVIEW

File: pandas/tests/arrays/test_boolean.py [hunk 1]
```diff
@@ -248,7 +248,11 @@ def test_coerce_to_numpy_array():
     tm.assert_numpy_array_equal(result, expected)
     # with missing values will raise error
     arr = pd.array([True, False, None], dtype="boolean")
-    with pytest.raises(ValueError):
+    msg = str(
+        "cannot convert to 'bool'-dtype NumPy array with missing values."
+        + " Specify an appropriate 'na_value' for this dtype."
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/test_boolean.py]: don't need the plus, string literals will be concatenated by the Python parser
```suggestion
        " Specify an appropriate 'na_value' for this dtype."
```

File: pandas/tests/arrays/test_boolean.py [hunk 2]
```diff
@@ -248,7 +248,11 @@ def test_coerce_to_numpy_array():
     tm.assert_numpy_array_equal(result, expected)
     # with missing values will raise error
     arr = pd.array([True, False, None], dtype="boolean")
-    with pytest.raises(ValueError):
+    msg = str(
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/test_boolean.py]: don't need the string constructor
```suggestion
    msg = (
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @ShilpaSugan generally lgtm. I think we have a check that whitespace is at the end of the strings, so if the CI fails may need to change that.

File: pandas/tests/arrays/test_boolean.py (hunk 1 above)
Comment (ShilpaSugan) [reply pandas/tests/arrays/test_boolean.py]: Thanks for the suggestion! I have made the changes.

File: pandas/tests/arrays/test_boolean.py (hunk 2 above)
Comment (ShilpaSugan) [reply pandas/tests/arrays/test_boolean.py]: Thanks for the suggestion! I have made the changes.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @ShilpaSugan lgtm pending green

**Pull request merged**